### PR TITLE
Add conformance tests for signal events

### PR DESCRIPTION
**Pull request by CTI777**
Related to https://github.com/Activiti/Activiti/issues/2662

### CODE REVIEW

File: activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/process/ProcessRuntimeBPMNSignalReceivedIT.java [hunk 1]
```diff
@@ -189,4 +190,28 @@ public void shouldGetSignalReceivedEventWithVariables() {
...
+
+        //when
+        ProcessInstance process = processRuntime.start(ProcessPayloadBuilder.start()
+                                                               .withProcessDefinitionKey("signalThrowEventProcess")
+                                                               .withVariable("name",
+                                                                             "peter")
+                                                               .build());
+        
+        //then
+        assertThat(process.getStatus()).isEqualTo(ProcessInstanceStatus.COMPLETED);
```
Comment (salaboy) [inline activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/process/ProcessRuntimeBPMNSignalReceivedIT.java]: @CTI777 @erdemedeiros what is this test testing?
We should at least check that the BPMN elements (events) for the throw signal is emitted right?

File: activiti-spring-conformance-tests/activiti-spring-conformance-signals/src/test/java/org/activiti/spring/conformance/signals/SignalThrowCatchTest.java [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+        securityUtil.logInAs("user1");
+
+        String processInstanceId = startIntermediateCatchEventSignalProcess();
+
+        SignalPayload signalPayload = ProcessPayloadBuilder.signal()
+                .withName("Test")
+                .withVariable("signal-variable",
+                              "test")
+                .build();
+        processRuntime.signal(signalPayload);
```
Comment (salaboy) [inline activiti-spring-conformance-tests/activiti-spring-conformance-signals/src/test/java/org/activiti/spring/conformance/signals/SignalThrowCatchTest.java]: @CTI777 this test will be more clear if we split the operations and the assertions in half.. 
1) Start and the assertions
   1.1) We can check that the last event here is the BPMN element that is going to catch the signal. 
2) Signal and the assertions

File: activiti-spring-conformance-tests/activiti-spring-conformance-signals/src/test/java/org/activiti/spring/conformance/signals/SignalThrowCatchTest.java [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+        collectedEvents.clear();
+    }
+
+    @SuppressWarnings("unused")
+    @Test
+    public void testProcessesWithThrowCatchSignal() {
+        securityUtil.logInAs("user1");
+
+        String processInstanceCatch = startIntermediateCatchEventSignalProcess();
+        String processInstanceThrow = startThrowSignalProcess();
```
Comment (salaboy) [inline activiti-spring-conformance-tests/activiti-spring-conformance-signals/src/test/java/org/activiti/spring/conformance/signals/SignalThrowCatchTest.java]: @CTI777 same here.. let's split the operations and the assertions in two..

File: activiti-spring-conformance-tests/activiti-spring-conformance-signals/src/test/java/org/activiti/spring/conformance/signals/SignalThrowCatchTest.java [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+        securityUtil.logInAs("user1");
+
+        String processInstanceId = startBoundaryEventSignalProcess();
+
+        SignalPayload signalPayload = ProcessPayloadBuilder.signal()
+                .withName("Test")
+                .withVariable("signal-variable",
+                              "test")
+                .build();
+        processRuntime.signal(signalPayload);
```
Comment (salaboy) [inline activiti-spring-conformance-tests/activiti-spring-conformance-signals/src/test/java/org/activiti/spring/conformance/signals/SignalThrowCatchTest.java]: @CTI777 same here

File: activiti-spring-conformance-tests/activiti-spring-conformance-signals/src/test/java/org/activiti/spring/conformance/signals/SignalThrowCatchTest.java [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+    @Test
+    public void testProcessStartedBySignal() {
+        securityUtil.logInAs("user1");
+
+        SignalPayload signalPayload = ProcessPayloadBuilder.signal()
+                .withName("SignalStart")
+                .withVariable("signal-variable",
+                              "test")
+                .build();
+        processRuntime.signal(signalPayload);
```
Comment (salaboy) [inline activiti-spring-conformance-tests/activiti-spring-conformance-signals/src/test/java/org/activiti/spring/conformance/signals/SignalThrowCatchTest.java]: @CTI777 same here

**Comment by CTI777** [comment]
@salaboy @erdemedeiros Refactoring for tests is made

**Pull request merged**